### ISSUE TITLE
Allow simple tiled and tabbed layouts

### ISSUE THREAD

**Issue by savchenko** [ammgws/autotiling-rs#16]
First thing first - excellent tool, does what it says on the packaging.

Would it be possible to limit nesting at a value > 0? This would help with situations like this one:

![image](https://github.com/ammgws/autotiling-rs/assets/300146/9f29a447-97a5-4746-81bc-f38d0beee46b)

Ideally, I would love Sway to:

- Allow switching between tiled and tabbed layouts.
- Avoid parent/child nonsense (`[H] / [V]` containers, nested levels, etc).

Is it possible without https://github.com/i3/i3/issues/3001 fixed upstream? That issue was opened back in 2017...

**Comment by ammgws** [comment]
I don't think we can do this until sway implements the equivalent of https://github.com/i3/i3/pull/5469

**Comment by MartenBE** [comment]
> I don't think we can do this until sway implements the equivalent of [i3/i3#5469](https://github.com/i3/i3/pull/5469)

This has been merged, can this be revisited?

**Comment by ammgws** [comment]
It was merged in August so you can try sway from master branch or wait until next release to see if it helps with this.

**Comment by savchenko** [comment]
I've been running that patch applied on top of 1.11 (`sway version 1.11-990b23c0`), it doesn't quite help.

Having an option to limit the nesting would still be helpful.

**Comment by pagesStartBlank** [comment]
I don't really know if this is what you are looking for but you can check out ... https://github.com/pagesareblank/autotiling-rs/tree/depth-limit

**Comment by savchenko** [comment]
@pagesareblank , looks promising, however `./autotiling-rs -w 9 -l 0` still allows to create this monstrosity:

<img width="562" height="46" alt="Image" src="https://github.com/user-attachments/assets/023bc873-d107-4025-a7a0-7bd558cf14cf" />

Compiled from your branch, 0c9a06e8951d6462f0cced0a195ed838150889cb

Sway 1.10.1
Debian 13.6

**Comment by pagesStartBlank** [comment]
@savchenko I have played around with that and while it is fairly easy to flatten it out and make a normal tabbed layout, putting it back together has become quite the undertaking. We will see where this takes me..... no promises

**Comment by savchenko** [comment]
@pagesareblank , nothing special on my side. I have `bindsym $mod+Return exec exec $term` in the Sway config as was essentially <kbd>Mod</kbd>+<kbd>Enter</kbd> interleaved with <kbd>Mod</kbd>+<kbd>-</kbd> (horizontal split) and <kbd>Mod</kbd>+<kbd>|</kbd> (vertical split).

I would wish for this nesting to be limited at the given level. Visually:

### This is desirable

<img width="1926" height="1072" alt="Image" src="https://github.com/user-attachments/assets/76147874-9f6e-4dce-949b-33c0d101adbb" />

### This is undesirable

<img width="1927" height="1071" alt="Image" src="https://github.com/user-attachments/assets/cfe0b497-97b9-458b-8af6-ad4712b677a2" />

**Comment by pagesStartBlank** [comment]
@savchenko Ok i believe i fixed this I am not too well versed in Rust so i wrote this in C hope this was what you where looking for .... https://github.com/pagesareblank/Swaytile

**Comment by savchenko** [comment]
@pagesareblank , I am starting to suspect that something is wrong on my end... `./swaytile -l 1 -v`:

<img width="480" alt="Image" src="https://github.com/user-attachments/assets/5f124d80-a1e4-48a4-8b53-fbf4bb796f33" />

Dead standard Sway installed from Debian stable. Any pointers?

**Comment by pagesStartBlank** [comment]
@savchenko is this with the toggle function ? or is this plain tiling, like do you switch between layer modes here ?

**Comment by pagesStartBlank** [comment]
It should look a little more like this 

<img width="1910" height="1048" alt="Image" src="https://github.com/user-attachments/assets/3c3e452b-e24b-4f50-9e63-19a56d36e0fd" />

or this when tabbed 

<img width="1901" height="1034" alt="Image" src="https://github.com/user-attachments/assets/62659afb-9db9-4fb2-93e2-d2c58cff1ca0" />

**Comment by pagesStartBlank** [comment]
and not necceserrily a problem on your end the deep nesting was definetly a problem with standart Sway layer tabbed and stacked, but my cosutum keybinds should remidy this 

<img width="1896" height="1036" alt="Image" src="https://github.com/user-attachments/assets/4a1dc90d-1cc3-4b5c-b22d-1dfd95baca58" />
